djstripe/models.py

```diff
@@ -541,20 +542,95 @@ def _attach_objects_hook(self, cls, data):
         if subscription:
             self.subscription = subscription
 
+    def _attach_objects_post_save_hook(self, cls, data):
+        # InvoiceItems need a saved invoice because they're associated via a
+        # RelatedManager, so this must be done as part of the post save hook.
+        cls._stripe_object_to_invoice_items(InvoiceItem, data, self)
+
+    @property
+    def plan(self):
+        """ Gets the associated plan for this invoice.
+
+        In order to provide a consistent view of invoices, it is necessary to
+        extract it from the first invoice item with a plan, rather than getting
+        it from the subscription by foreign key.
```

Comment:
@lskillen: I should clarify: What is the subject in this sentence? What are you extracting? The plan ID?